inst/include/common/fims_vector.hpp

```diff
@@ -0,0 +1,483 @@
+     */
+    reverse_iterator rend() {
+        return this->vec_m.rend();
+    }
+    
+    
+    /**
+     * @brief Returns a constant reverse iterator to the first element of the reversed vector. It corresponds to the last element of the non-reversed vector.
+     */
+    const_reverse_iterator rbegin() const {
+        return this->vec_m.rbegin();
+    }
+    
+    
+    /**
+     * @brief Returns a constant reverse iterator to the element following the last element of the reversed vector. It corresponds to the element preceding the first element of the non-reversed vector.
+     */
+    const_reverse_iterator rend() const {
+        return this->vec_m.rend();
+    }
+    
+    //capacity
+    
+    
+    /**
+     * @brief Checks whether the container is empty.
+     */
+    bool empty() {
+        return this->vec_m.empty();
+    }
+    
+    /**
+     * @brief Returns the number of elements.
+     */
+    size_type size() const {
+        return this->vec_m.size();
+    }
+    
+    /**
+     * @brief Returns the maximum possible number of elements.
+     */
+    size_type max_size() const {
+        return this->vec_m.max_size();
+    }
+    
+    /**
+     * @brief Reserves storage.
+     */
+    void reserve(size_type cap) {
+        this->vec_m.reserve(cap);
+    }
+    
+    
+    /**
+     * @brief Returns the number of elements that can be held in currently allocated storage.
+     */
+    size_type capacity() {
+        return this->vec_m.capacity();
+    }
+    
+    /**
+     *  @brief Reduces memory usage by freeing unused memory.
+     */
+    void shrink_to_fit() {
+        this->vec_m.shrink_to_fit();
+    }
+    
+    //modifiers
+    
+    /**
+     * @brief Clears the contents.
+     */
+    void clear() {
+        this->vec_m.clear();
+    }
+    
+    
+    /**
+     * @brief Inserts value before pos.
+     */
+    iterator insert(const_iterator pos, const Type& value) {
+        return this->vec_m.insert(pos, value);
+    }
+    
+    /**
+     * @brief Inserts count copies of the value before pos.
+     */
+    iterator insert(const_iterator pos, size_type count, const Type& value) {
+        return this->vec_m.insert(pos, count, value);
+    }
+    
+    /**
+     * @brief Inserts elements from range [first, last) before pos.
+     */
+    template< class InputIt >
+    iterator insert(const_iterator pos, InputIt first, InputIt last) {
+        return this->vec_m.insert(pos, first, last);
+    }
+    
+    /**
+     * @brief Inserts elements from initializer list ilist before pos.
+     */
+    
+    iterator insert(const_iterator pos, std::initializer_list<Type> ilist) {
+        return this->vec_m.insert(pos, ilist);
+    }
+    
+    
+    /**
+     * @brief Constructs element in-place.
+     */
+    template< class... Args >
```

Comment:
Do we expect this to be used in FIMS? I have not seen the `class...` syntax or `Args&&...` syntax and would prefer not to introduce this syntax unless we expect to use this function